lib/central_mail/utilities.rb

```diff
@@ -48,20 +48,19 @@ def log_submission(uploaded_object, metadata)
 
     def retry_errors(e, uploaded_object)
       if e.code == 'DOC201' && @retries <= RETRIES
-        self.class.perform_at(30.minutes.from_now, uploaded_object.id, { caller: 'DOC201_Retry' }, @retries + 1)
+        self.class.perform_at(30.minutes.from_now, uploaded_object.guid, { caller: 'DOC201_Retry' }, @retries + 1)
       else
         uploaded_object.update(status: 'error', code: e.code, detail: e.detail)
       end
       log_error(e, uploaded_object)
     end
 
     def log_error(e, uploaded_object)
-      uuid = uploaded_object.respond_to?(:guid) ? uploaded_object.guid : uploaded_object.id
```

Comment:
Since `retry_errors` (and by extension, `log_error`) is only called after the `UploadSubmission` record is looked up using its GUID (in `UploadProcessor`), we can be sure that it always has one.